### PR TITLE
Upgrade to Go 1.18 and to latest dependencies

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

# Changes

- Upgrade to go 1.18
- Upgrade to the latest dependencies

**Release Note**

```release-note
- 🧽 Bump go version to 1.8
- 🧽 Bump k8s version to 1.25.2 
```

### CODE REVIEW

**Comment by pierDipi** [comment]
@ricardozanini there are build tests failing with:
```
home/prow/go/src/knative.dev/eventing-kogito is out of date. Please run hack/update-codegen.sh
```

**Comment by ricardozanini** [comment]
> @ricardozanini there are build tests failing with:
> 
> ```
> home/prow/go/src/knative.dev/eventing-kogito is out of date. Please run hack/update-codegen.sh
> ```

Yes, I saw that yesterday. The problem is that I've run it multiple times. I'll try to figure out what's happening there.

File: go.mod [hunk 1]
```diff
@@ -1,16 +1,105 @@
...
+	k8s.io/kube-openapi v0.0.0-20220803162953-67bda5d908f1 // indirect
+	k8s.io/utils v0.0.0-20220728103510-ee6ede2d64ed // indirect
+	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 replace github.com/prometheus/client_golang => github.com/prometheus/client_golang v0.9.2
+
+replace k8s.io/api => k8s.io/api v0.25.2
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply go.mod:97]: It was added by the update-k8s-deps script. I'll manually remove them.

[the remark this answers is not in the document]
Comment (ricardozanini) [reply go.mod:97]: Locally I'm using 1.19, hence the formatting problem. Thanks @lionelvillard for spoting this 👍

**Pull request merged**